defaults/test_default.yaml

```diff
@@ -241,3 +241,6 @@ validate_large_collections: false
 scylla_d_overrides_files: []
 
 run_commit_log_check_thread: true
+
+# teardown validators
+teardown_validate_sstables: true
```

Comment:
Having a boolean configuration might a bit naive

Maybe we should consider things like percentage of stable to scrub ?

Also maybe a configuration which is a list of possible validators to use, and not introduce multiple top level SCT configurations for each validator ? (I know it's the first one we introduce, but there would be more that we should be thinking of)

Reply:
> Having a boolean configuration might a bit naive

what else do we need here? Timeout?

> Maybe we should consider things like percentage of stable to scrub ?

I don't know the way to take percentage

> Also maybe a configuration which is a list of possible validators to use, and not introduce multiple top level SCT configurations for each validator ? (I know it's the first one we introduce, but there would be more that we should be thinking of)

I was thinking about that and was not sure, I took these factors when making decision:
1. when one test decides on disabling this one, then overriding one configuration would mean overriding all validators configuration if we used one config for all
2. different validators might need different configuration, so we are not stuck to naive config like here
3. putting in the same area in `test_default.yaml` will make easy to find possible validators (we could include also ones that are disabled by default)

But I'm open to different arugments :)

Reply:
> > Having a boolean configuration might a bit naive
> 
> what else do we need here? Timeout?
> 
> > Maybe we should consider things like percentage of stable to scrub ?
> 
> I don't know the way to take percentage

don't you feed it sstables one by one ? I guess you could random select and feed the tool only part of the sstables ?

> 
> > Also maybe a configuration which is a list of possible validators to use, and not introduce multiple top level SCT configurations for each validator ? (I know it's the first one we introduce, but there would be more that we should be thinking of)
> 
> I was thinking about that and was not sure, I took these factors when making decision:
> 
> 1. when one test decides on disabling this one, then overriding one configuration would mean overriding all validators configuration if we used one config for all

that's not always try, for example see the what we use `stress_image`, we can replace specific one, without touch others, i.e. we can extend/overwrite inner structures. 

> 2. different validators might need different configuration, so we are not stuck to naive config like here

that's for sure is correct, so something similar to how `run_fullscan` is being used, i.e. with converting it to a dataclass
we know we gonna have parameters for each one, we should dine it as such from start

> 3. putting in the same area in `test_default.yaml` will make easy to find possible validators (we could include also ones that are disabled by default)

I don't think we gonna have any of them in test_default.yaml, not for a very long time, until we prove each one of them, we won't have them by default for all tests.

> 
> But I'm open to different arugments :)

Reply:
Ok, I agree I'll adapt accordingly. I'll check how `stress_image` works as I don't know yet mechanism behind merging dicts by sct_config. Anyway assuming these won't be defaults, we can specify test by test what validators are used.

As to scrub validation, we don't provide sstables, we can provide keyspace - this will be added to config param.

> Agree with @fruch Example of the new config option I would like to see:
> 
> ```
> teardown_validators:
>   nodetool_scrub: '{"mode": "VALIDATE", "timeout_per_node": 1200}'
> ```

I disagree to provide "mode" in config - we always want to validate. In case we'll need different modes, we'll add this.

Reply:
We can have a percentage for the upload of broken stables, if we find uploading all of them is too big

Reply:
> We can have a percentage for the upload of broken stables, if we find uploading all of them is too big

Sounds like an good idea, can we do it in followup task?
Also would require some more thinking - e.g. what are the thresholds for selecting percentage and what percentage.

Reply:
Yes it can be a followup, it was just used as example of possible configuration we might want to add